libkodimote/imagecache.cpp

```diff
@@ -173,6 +174,12 @@ void KodiImageCache::imageFetched()
         }
     }
 
+    foreach (const ImageFetchJob::Callback callback, job->callbacks()) {
+        QMetaObject::invokeMethod(callback.object().data(), callback.method().toLatin1(), Qt::QueuedConnection, Q_ARG(int, job->id()));
+    }
+
+    job->deleteLater();
```

Comment:
I think this shouldn't be needed, as the `QNetworkRequest` takes ownership of the "originatingObject", that's why the "success branch" reparents it again to the ImageCache. So when the reply gets deleted, it also deletes the request, and thus also the job.

Besides invoking the callbacks seems a bit odd to me. Same for setting the image presence to true, with an empty cache file. I think it would be better to leave `m_cacheFiles` as is, so a new `contains` call doesn't have to check if it exists, and downloading (/preparing) is retried because it doesn't exist.